movement/movement.scm

```diff
@@ -0,0 +1,47 @@
+;
+; ROS robot movement module.
+;
+; Provides interfaces for making physical movements, and
+; for saying things.
+
+(define-module (opencog movement))
+
+(use-modules (opencog) (opencog atom-types) (opencog python))
+
+(load "movement/do-movement.scm")
+
+; Try loading the python code from this directory; else got for the
+; install directory.  This assumes that the the current directory
+; is in the python sys.path.
+;
+; If roscore is not running, then the load will hang. Thus, to avoid the
+; hang, we test to see if we can talk to roscore. If we cannot, then load
+; only the debug interfaces.
+;
+;
+(define-public start-ros-movement-node
+	(python-eval "
+import os.path
+import sys
+import rosgraph
+# This hard-coded install path from CMakefile
+sys.path.append('/usr/local/share/opencog/python')
+try:
+    # Throw an exception if roscore is not running.
+    rosgraph.Master('/rostopic').getPid()
+    if (os.path.isfile('atomic.py')):
+        # Python3 does not support execfile any longer
+        # execfile('atomic.py'))
+        exec(open('atomic.py').read())
```

Comment:
yes, ros is currently python2 only, but I figure that having code that works with either version would be better.